### PR TITLE
Fix #97: Populate property bags w/ pass-through key names

### LINKED ISSUE #97: property bag s/be populated with pass-through key names
I have changed my opinion on how property bags should be populated. Previously, I stated that key names should be converted to camel-case in order to make the content more JSON friendly. Having looked at the way several tools populate this additional data, however, I have changed my mind. (The specific example that prompted this issue is the clang analysis converter, which populates the property bag with key names such as issue_hash. We don't use underscores elsewhere in the format, so this raises the question of whether the key name should be normalized in some way).

At this point, I tend to think that converters should preserve the casing of additional content they add to the property bag. It is difficult to contrive a naming standard that works for everyone and the benefit is not clear (while putting more burden on converter authors). By not altering these values, it makes it easier to match this data with source data from the native tool format. 

The fact that these names jump out as distinct from the rest of the format is arguably fine, since the information isn't, in fact, part of the standard.

All of the above is in support of a general principle I think we've landed on that converters should generally be a pass-through/direct conversion operation and minimize normalization or other transformations.

### PR DESCRIPTION
**Pull request by ghost**
@michaelcfanning

### CODE REVIEW

File: document/AnnexC.mdk [hunk 1]
```diff
@@ -12,9 +12,16 @@ purely to express that guidance.
 
 Converters should populate those elements of the SARIF format
```
Comment (ghost) [inline document/AnnexC.mdk:13]: Annex C is the "guidance for converters" section.

File: Static Analysis Results Interchange Format (SARIF).html [hunk 2]
```diff
@@ -2431,25 +2431,32 @@ <h2 id="result-management-profile"  data-line="34;document\ResultManagementProfi
...
-If the input data includes information for which there is no SARIF equivalent,
+</p>
+<p class="indent" data-line="46;document\AnnexC.mdk:16"><span data-line="46;document\AnnexC.mdk:16"></span>If the input data includes information for which there is no SARIF equivalent,
 converters may use it to populate the various property bags and tag lists defined by the SARIF format,
 or they may simply omit it from the output.
-If the input data does not include an equivalent for any SARIF element,
-the converter should <span data-line="46;document\AnnexC.mdk:19"></span><em>not</em><span data-line="46;document\AnnexC.mdk:19"></span> attempt to synthesize that element.
+When populating a property bag with such information,
+converters should use a property name that matches the name of that piece of information in the
+native tool format (assuming of course that that name is a valid JSON property name),
```
Comment (michaelcfanning) [inline Static Analysis Results Interchange Format (SARIF).html]: do you have guidance for the condition where a property name is not a valid JSON property name?

Comment (ghost) [inline Static Analysis Results Interchange Format (SARIF).html]: Actually come to think of it, a property name in JSON is grammatically just a string, and a JSON string can contain any character, up to and including the empty string -- but some characters need to be escaped. I'll mention that.

**Comment by michaelcfanning** [comment]
LGTM

**Pull request merged**